org.eclipse.jdt.core/search/org/eclipse/jdt/internal/core/search/indexing/SourceIndexer.java

```diff
@@ -208,6 +215,35 @@ private void purgeMethodStatements(TypeDeclaration type) {
 				purgeMethodStatements(memberTypes[i]);
 	}
 
+	private boolean definesAnonymousType(AbstractMethodDeclaration method) {
```

Comment:
The code here is IMO bad, but I don't know how else to write the code. The types in the CU are not yet resolved at this point, so (I think) we can't depend on more reliable APIs to find anonymous/inner types.